avid-adventurers/src/pages/survey/Rate.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { Label } from '../../components/Label/Label';
+import Rating from '../../components/Rating/Rating';
+import { Container, Title, FormWrapper, ButtonRow, BackButton, ContinueButton } from '../styles';
+import { useSurvey } from '../../utils/surveyContext';
+import { LabeledInput } from '../../components/LabeledInput/LabeledInput';
+
+
+const Rate: React.FC = () => {
+    const navigate = useNavigate();
+    const [rating, setRating] = useState(0);
```

Comment:
This should not be a new hook, but rather inherit from the surveyContext that you defined earlier. Since the rating is saved as a global context, you shouldn't need to save it into localStorage every time it's changed.

Reply:
resolved